src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -343,20 +383,23 @@ public InventorySyncStatistics getStatistics() {
     }
 
     /**
-     * Returns {@link CompletionStage} instance which may contain mapping of {@link ChannelKeyExtractor} to
-     * {@link InventoryEntry} of instances existing in a CTP project. Instances are fetched from API by skus, that
-     * corresponds to skus present in {@code drafts}. If fetching existing instances results in exception then
-     * returned {@link CompletionStage} contains such exception.
+     * Tries to fetch from a CTP a list of inventory entries of SKUs present in {@code drafts}.
+     * If operation succeed then {@link CompletionStage} containing {@link Optional} with fetched {@link List} is
+     * returned, otherwise error callback function is executed and {@link CompletionStage} with empty {@link Optional}
+     * is returned.
      *
      * @param drafts {@link List} of drafts
-     * @return {@link CompletionStage} instance which contains either {@link Map} of {@link ChannelKeyExtractor} to
-     *      {@link InventoryEntry} of instances existing in a CTP project that correspond to passed {@code drafts} by
-     *      sku comparision, or exception occurred during fetching existing inventory entries
+     * @return {@link CompletionStage} instance with {@link Optional} that may contain list of inventory entries
      */
-    private CompletionStage<List<InventoryEntry>> fetchExistingInventories(@Nonnull final List<InventoryEntryDraft>
-                                                                               drafts) {
+    private CompletionStage<Optional<List<InventoryEntry>>> fetchExistingInventories(@Nonnull final
+                                                                                     List<InventoryEntryDraft> drafts) {
         final Set<String> skus = extractSkus(drafts);
-        return inventoryService.fetchInventoryEntriesBySkus(skus);
+        return inventoryService.fetchInventoryEntriesBySkus(skus)
+            .thenApply(Optional::of)
```

Comment:
exactly, I don't mean to change the position of the `exceptionally` block, it's in the correct place :)

Not sure how is `optional` better in this case. 

This is what I mean:

````java
// I would even name this fetchOldEntriesAndSync()
private CompletionStage<Void> processBatch(@Nonnull final List<InventoryEntryDraft> batchOfDrafts) {
    return fetchExistingInventories(batchOfDrafts)
        .thenCompose(oldInventories -> syncBatch(oldInventories, batchOfDrafts));
     }

private CompletionStage<List<InventoryEntry>> fetchExistingInventories(@Nonnull final List<InventoryEntryDraft> drafts) {
    final Set<String> skus = extractSkus(drafts);
    return inventoryService.fetchInventoryEntriesBySkus(skus)
        .exceptionally(exception -> {
            final String errorMessage = format(CTP_INVENTORY_FETCH_FAILED, extractSkus(drafts));
            handleError(errorMessage, exception, drafts.size());
            return null;
            });
    }

````

Why not do it this way?